### PR TITLE
[CoreML EP] Add Gemm/MatMul support

### PR DESCRIPTION
**Pull request by gwang-msft**
**Description**: [CoreML EP] Add Gemm/MatMul support

**Motivation and Context**
- Fill the op support gap

### CODE REVIEW

File: onnxruntime/core/providers/coreml/builders/impl/builder_utils.cc [hunk 1]
```diff
@@ -80,18 +82,20 @@ common::Status HandleAutoPad(const std::vector<int64_t> input_shape,
   return Status::OK();
 }
 
+void CreateCoreMLWeight(CoreML::Specification::WeightParams& weight,
+                        const float* data, size_t num_elements) {
+  weight.mutable_floatvalue()->Clear();
+  std::copy(data, data + num_elements,
+            google::protobuf::RepeatedFieldBackInserter(weight.mutable_floatvalue()));
+}
```
Comment (skottmckay) [inline onnxruntime/core/providers/coreml/builders/impl/builder_utils.cc:90]: Is it possible to use Reserve and copy_n, or alternatively call Reserve prior to the copy? Not a big deal thought given this is during setup.

File: onnxruntime/core/providers/coreml/builders/impl/gemm_op_builder.cc [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+// Add operator related
+
+void GemmOpBuilder::AddInitializersToSkip(ModelBuilder& model_builder, const Node& node) const {
+  const auto& op = node.OpType();
+  const auto& input_defs(node.InputDefs());
+  model_builder.AddInitializerToSkip(input_defs[1]->Name());
+  if (op == "Gemm" && input_defs.size() > 2) {
+    model_builder.AddInitializerToSkip(input_defs[2]->Name());
+  }
+}
```
Comment (skottmckay) [inline onnxruntime/core/providers/coreml/builders/impl/gemm_op_builder.cc:41]: Comment would be good saying why we add these to the skipped list.

File: onnxruntime/core/providers/coreml/builders/impl/builder_utils.cc (hunk 1 above)
Comment (gwang-msft) [reply onnxruntime/core/providers/coreml/builders/impl/builder_utils.cc:90]: Tried reserve and copy_n, seems not working (UT failure), added reserve before the copy

Comment (skottmckay) [reply onnxruntime/core/providers/coreml/builders/impl/builder_utils.cc:90]: Sorry - mean Resize + copy_n. But Reserve is fine too - just may not translate to a block copy but at least will only have one memory allocation.

**Review by skottmckay — DISMISSED** [review]
:shipit:

File: onnxruntime/core/providers/coreml/builders/impl/builder_utils.cc (hunk 1 above)
Comment (gwang-msft) [reply onnxruntime/core/providers/coreml/builders/impl/builder_utils.cc:88]: Switched to list initialization

**Review by skottmckay — APPROVED** [review]
:shipit:

**Pull request merged**